src/main/java/org/broadinstitute/dsm/util/DDPKitRequest.java

```diff
@@ -169,4 +198,100 @@ public void requestAndWriteKitRequests(List<LatestKitRequest> latestKitRequests)
+                                                    null, null, rs.getString(DBConstants.EXTERNAL_ORDER_NUMBER), ddpParticipant,
+                                                    rs.getString(DBConstants.EXTERNAL_ORDER_STATUS),
+                                                    rs.getString("subkits." + DBConstants.EXTERNAL_KIT_NAME),
+                                                    rs.getLong(DBConstants.EXTERNAL_ORDER_DATE)));
+                                            logger.info("Added unordered kit with external order number " + orderKit.get(orderKit.size() - 1).getExternalOrderNumber() + " to the order list");
+                                        }
+
+                                    }
+
+                                }
+                            }
+
+                        }
+                        kitsToOrder.put(kitRequestSettings,orderKit);
+                    }
+                    catch (SQLException ex) {
+                        throw new RuntimeException("Can not query for unordered kits for external shipper "+ ex);
+                    }
+
+                }
+                catch (SQLException ex) {
+                    dbVals.resultException = ex;
+                }
+                return dbVals;
+            });
+            if(results.resultException != null){
+                throw new RuntimeException("Error getting approved kits for external shipper "+ results.resultException);
+            }
+
+        }
+    }
+
+        private void addSubKits (@NonNull List < KitSubKits > subKits, @NonNull KitDetail kitDetail, @NonNull String collaboratorParticipantId,
+                @NonNull KitRequestSettings kitRequestSettings, @NonNull String instanceId){
+            int subCounter = 0;
+            String externalOrderNumber = null;
+            if (StringUtils.isNotBlank(kitRequestSettings.getExternalShipper())) {
+                externalOrderNumber = generateExternalOrderNumber();
+            }
+            for (KitSubKits subKit : subKits) {
+                for (int i = 0; i < subKit.getKitCount(); i++) {
+                    //kitRequestId needs to stay unique -> add `_[SUB_COUNTER]` to it
+                    KitRequestShipping.addKitRequests(instanceId, subKit.getKitName(), kitDetail.getParticipantId(),
+                            subCounter == 0 ? kitDetail.getKitRequestId() : kitDetail.getKitRequestId() + "_" + subCounter, subKit.getKitTypeId(), kitRequestSettings,
+                            collaboratorParticipantId, kitDetail.isNeedsApproval(), externalOrderNumber);
+                    subCounter = subCounter + 1;
+                }
+            }
+        }
+
+        public static String generateExternalOrderNumber () {
+            String externalOrderNumber = NanoIdUtils.randomNanoId(
+                    NanoIdUtils.DEFAULT_NUMBER_GENERATOR, "1234567890QWERTYUIOPASDFGHJKLZXCVBNM".toCharArray(), 20);
```

Comment:
I saw that somewhere before. please use the same method here

Reply:
that is in ups which is not working yet. i'll use same method for that when I get to making that work